content/03-reference/01-tools-and-interfaces/02-prisma-client/05-filtering.mdx

```diff
@@ -0,0 +1,323 @@
+    },
+    published: {
+      equals: false,
+    },
+  },
+})
+```
+
+Retrieve all `Post` records where the `title` field contains `Prisma` or `databases`:
+
+```javascript
+const result2 = await prisma.post.findMany({
+  where: {
+    OR: [
+      {
+        title: {
+          contains: 'Prisma',
+        },
+      },
+      {
+        title: {
+          contains: 'databases',
+        },
+      },
+    ],
+  },
+})
+```
+
+Retrieve all `Post` records where the `title` field contains `Prisma` or `databases`, and `published` is `false`:
+
+```javascript
+const result = await prisma.post.findMany({
+  where: {
+    OR: [
+      {
+        title: {
+          contains: 'Prisma',
+        },
+      },
+      {
+        title: {
+          contains: 'databases',
+        },
+      },
+    ],
+    AND: {
+      published: false,
+    },
+  },
+})
+```
+
+Retrieve all `Post` records where the `title` filed contains `Prisma` or `databases`, but not `SQL`:
+
+```javascript
+const result2 = await prisma.post.findMany({
+  where: {
+    OR: [
+      {
+        title: {
+          contains: 'Prisma',
+        },
+      },
+      {
+        title: {
+          contains: 'databases',
+        },
+      },
+    ],
+    NOT: {
+      title: {
+        contains: 'SQL',
+      },
+    },
+  },
+})
+```
+
+You can [enable logging](logging) to debug complex queries:
+
+```javascript
+const prisma = new PrismaClient({
+  log: [
+    {
+      emit: 'event',
+      level: 'query',
+    },
+  ],
+})
+
+prisma.on('query', e => {
+  e.query, console.log(e)
+})
+```
+
+## Filter on related records
+
+You can also filter based on related record fields. For example, the following query returns all `Post` records where the user's email address is `sarah@prisma.io`:
+
+```javascript
+const result = await prisma.post.findMany({
+  where: {
+    user: {
+      email: {
+        equals: 'sarah@prisma.io',
+      },
+    },
+  },
+})
+```
+
+Prisma Client offers the following operators for filtering on one-to-many relationships:
+
+| Operator | Description                                                                       |
+| -------- | --------------------------------------------------------------------------------- |
+| `some`   | Returns all records where one or more _related_ records match filtering criteria. |
+| `every`  | Returns all records where all _related_ records match filtering criteria.         |
+| `none`   | Returns all records where no _related_ records match filtering criteria.          |
+
+### Examples
+
+Retrieve all `User` records where:
+
+- _All_ posts are published
+- At least one related `Post` mentions `Prisma`
+
+```javascript
+  const record = await prisma.user.findMany({
+    where: {
+      post: {
+          every: {
+            published: true
+          }
+        some: {
+          content: {
+            contains: "Prisma"
+          }
+        }
+      }
+    }
+  }
+```
+
+Retrieve all `Post` records where:
```

Comment:
General note on these "example headlines". On other pages I typically made them bold and didn't add a colon to the end.

In general I actually think we should probably make them `H5` or `H6` elements though so that we're able to link to individual examples which currently isn't possible.

Reply:
Removed : and made into `h5`/`h6` on this page + added backlog item to update all examples.

Reply:
Done! ✅